R/pin_versions.R

```diff
@@ -165,6 +165,9 @@ version_setup <- function(board, name, new_version, versioned = NULL) {
     versions <- pin_versions(board, name)
     old_version <- versions$version[[1]]
     n_versions <- nrow(versions)
+    if (old_version == new_version) {
+      return(NULL)
+    }
```

Comment:
What do you think about this alternative instead @hadley?

Step 1: Here in `version_setup()` we return `NULL` when the new version is the same as the old version.